python/dask_cudf/dask_cudf/core.py

```diff
@@ -420,7 +420,7 @@ def struct(self):
 
 
 class Index(Series, dd.core.Index):
-    _partition_type = cudf.Index
+    _partition_type = cudf.Index  # type: ignore
```

Comment:
This change is not related to the PR, but was added to avoid local mypy failures...

```
python/dask_cudf/dask_cudf/groupby.py:381: error: Need type annotation for '_global_set' (hint: "_global_set: Set[<type>] = ...")
python/dask_cudf/dask_cudf/core.py:423: error: Incompatible types in assignment (expression has type "Type[Index]", base class "Series" defined the type as "Type[Series]")
Found 2 errors in 2 files (checked 122 source files)
```

Reply:
cc: @shwina for the mypy failure

Reply:
I have previously encountered cases where unrelated changes to the code allow mypy's engine to discover typing errors that it previously couldn't. This mypy error looks real, but I think adding this ignore is fine for now. I assume that making `dd.Index` inherit from `dd.Series` was done for convenience at some point, and refactoring the class hierarchy is the only correct way to make this go away but is out of scope for this PR.